### DISCUSSION TITLE
Fluent Migrator 3.2.1 How do I change the timeout on the migration runner to handle sql scripts that take too long to execute?

### DISCUSSION THREAD

**Discussion by josephbaggettgainsco** [fluentmigrator/fluentmigrator#1472] [Q&A] [answered]
How do I change the timeout on the migration runner to handle sql scripts that take too long to execute but are completely valid going through FluentMigrator 3.2.1?

        /// <summary>
        /// Configure the dependency injection services
        /// </summary>
        private static IServiceProvider CreateServices()
        {
            return new ServiceCollection()
             // Add common FluentMigrator services
             .AddFluentMigratorCore()
             .Configure<RunnerOptions>(cfg =>
              {
                  cfg.Profile = !String.IsNullOrEmpty(profile) ? profile : string.Empty; 
              })
             .ConfigureRunner(rb => rb
                 // Add SQLite support to FluentMigrator
                 .AddSqlServer2016()
                 // Set the connection string
                 .WithGlobalConnectionString(GetConnectionString())
                 
                 // Define the assembly containing the migrations
                 .ScanIn(typeof(EmptyDatabaseMigration).Assembly).For.Migrations())

             // Enable logging to console in the FluentMigrator way
             .AddLogging(lb => lb.AddFluentMigratorConsole())
             // Add AssemblySourceItem to run VersionTableInfo updates
             .AddSingleton<IAssemblySourceItem>(new AssemblySourceItem(Assembly.GetAssembly(typeof(EmptyDatabaseMigration))))
             // Build the service provider
             .BuildServiceProvider(false);

        }

        /// <summary>
        /// Update the database
        /// </summary>
        private static void UpdateDatabase(IServiceProvider serviceProvider)
        {

            // Instantiate the runner
            var runner = serviceProvider.GetRequiredService<IMigrationRunner>();

            // will create testdata
            if (allowDatabaseToBeRebuilt)
            {
                runner.MigrateDown(0);
            }

            if (migrationRollbackNumber.HasValue)
            {
                runner.MigrateDown(migrationRollbackNumber.Value);
            }
            else
            {
                //runner.RunnerContext.Timeout = 0;
                // Execute the migrations
                runner.MigrateUp();
            }
        }

I tried to set the RunnerContext.Timeout to 0, but it is not fixing the issue.  I can run these sql statements manually and they simply just run.

Joseph

**Comment by fubar-coder** [comment, marked as answer]
Did you try a timeout value of `-1`? IIRC, [Timeout.Infinite](https://docs.microsoft.com/en-us/dotnet/api/system.threading.timeout.infinite?view=net-5.0) (for example) is defined as `-1`.

**Reply by jzabroski** [reply]
@josephbaggettgainsco Converted this to a Discussion and labeled it Q&A and marked @fubar-coder answer as an answer.

**Reply by josephbaggettgainsco** [reply]
@fubar-coder @jzabroski What are you setting to to Timeout.Infinite? I see the runner.RunnerContext is marked as Obsolete. I see this method on the ConfigureRunner method .WithGlobalCommandTimeout(TimeSpan.Infinite)

What specifically needs to be set for FluentMigrator 3.2.1?

**Reply by jzabroski** [reply]
I think @fubar-coder 's point is you wrote `//runner.RunnerContext.Timeout = 0;` but you should write `//runner.RunnerContext.Timeout = -1;`.

`.WithGlobalCommandTimeout(TimeSpan.Infinite)` sounds like it should work - did you try it?

**Comment by TropinAlexey** [comment]
Hello everyone.
In version 3.3.2 `runner.RunnerContext` is marked as obsolete. Where can I found it now?

**Reply by jzabroski** [reply]
Use Dependency Injection to setup your context.

**Reply by oksankom** [reply]
Could you please clarify this? I see that the whole interface IRunnerContext is marked as obsolete. Some properties of it were moved to RunnerOptions, but Timeout was not and I can't find it anywhere.

**Reply by jzabroski** [reply]
https://github.com/fluentmigrator/fluentmigrator/blob/028340e1e670321d9e19619dccaf7735deeaae2e/src/FluentMigrator.Runner.Core/MigrationRunnerBuilderExtensions.cs#L94
